### ISSUE TITLE
cx_oracle emits deprecation warning and/or segfaults under python 3.8.0b1

### ISSUE THREAD

**Issue by zzzeek** [oracle/python-cx_Oracle#317]
cx_Oracle generates this warning under Python 3.8.0b1 when an output type handler is used:

    DeprecationWarning: PY_SSIZE_T_CLEAN will be required for '#' formats

If you turn on the "error" handler for the warnings filter, then it segfaults.

Complete example:

```
import cx_Oracle
import warnings

# comment to only emit a warning, not segfault
warnings.filterwarnings("error", category=DeprecationWarning)


conn = cx_Oracle.connect(
    user="scott",
    password="tiger",
    dsn=cx_Oracle.makedsn("oracle1120", 1521, sid="xe"),
)

def output_type_handler(
    cursor, name, default_type, size, precision, scale
):
    return None

# comment out to remove the warning
conn.outputtypehandler = output_type_handler

cursor = conn.cursor()

cursor.execute(
    "select 1 from dual",
)

```

**Comment by methane** [comment]
ref: https://bugs.python.org/issue36381

When parsing or building value by format with "#" (e.g. "s#"), Python will not support int for size from Python 3.10.  We need to use `Py_ssize_t` instead and `#define PY_SSIZE_T_CLEAN` before `#include <Python.h>`.

**Comment by anthony-tuininga** [comment]
Just to reassure you: I haven't lost track of this and have already made some of the necessary changes. I hope to have this addressed next week.

**Comment by anthony-tuininga** [comment]
This has been released with cx_Oracle 7.2.
